physics/gwdps.f

```diff
@@ -1237,17 +1244,31 @@ subroutine gwdps_run(                                             &
 !    &                      dbim,idxzb(I),U1(J,K),V1(J,K),me
 
             tem1     = DBIM * DEL(J,K)
-            DUSFC(J) = DUSFC(J) - tem1 * U1(J,K)
-            DVSFC(J) = DVSFC(J) - tem1 * V1(J,K)
+            DUSFC(J) = DUSFC(J) + onebg * tem1 * U1(J,K)
+            DVSFC(J) = DVSFC(J) + onebg * tem1 * V1(J,K)
+            ! Output blocking tendencies if ldiag_ugwp=.true.
+            if (ldiag_ugwp) then
+               dtaux2d_bl(j,k) = - DBIM * U1(J,K)
+               dtauy2d_bl(j,k) = - DBIM * V1(J,K)
+               dusfc_bl(j) = dusfc_bl(j) + dtaux2d_bl(j,k) * del(j,k)
```

Comment:
The unit for dtaux2d_bl is m s-2,  the unit for del is Pa, and the unit for dusfc_bl is Pa. So something is wrong here. Maybe dtaux2d_bl needs to be divided by g (9.81 m s-2)? e.g., dusfc_bl(j) = dusfc_bl(j) + dtaux2d_bl(j,k)/g * del(j,k)

Reply:
You are correct.  However, we take care of this a few lines below (we divide by g, or multiply by onebg):

1283       if (ldiag_ugwp) then
1284          !  Finalize dusfc and dvsfc diagnostics for gsl drag components
1285          DO i = 1,npt
1286             j = ipt(i)
1287             dusfc_ms(j) = - onebg * dusfc_ms(j)
1288             dvsfc_ms(j) = - onebg * dvsfc_ms(j)
1289             dusfc_bl(j) = - onebg * dusfc_bl(j)
1290             dvsfc_bl(j) = - onebg * dvsfc_bl(j)
1291          ENDDO
1292       end if

Reply:
I see. Thanks.